src/cgroup.rs

```diff
@@ -328,9 +414,22 @@ fn supported_controllers() -> Vec<String> {
         .collect::<Vec<String>>()
 }
 
-fn create_v2_cgroup(root: PathBuf, path: &str) -> Result<()> {
+fn create_v2_cgroup(
+    root: PathBuf,
+    path: &str,
+    specified_controllers: Option<Vec<String>>,
+) -> Result<()> {
     // controler list ["memory", "cpu"]
-    let controllers = supported_controllers();
+    let controllers = if let Some(s_controllers) = specified_controllers {
+        if verify_supported_controllers(s_controllers.as_ref()) {
+            s_controllers
+        } else {
+            return Err(Error::new(ErrorKind::SpecifiedControllers));
+        }
+    } else {
+        get_supported_controllers()
```

Comment:
If specified controllers contain supported + unsupported controllers, and specfied controllers does not contain all suppored controllers, does this create redundant controllers?

Reply:
When the specified controller contains an unsupported controller, an error SpecifiedControllers will be returned.
```
#[error("specified controller is not in the list of supported controllers")]
SpecifiedControllers,
```